### PR TITLE
(PC-31067) refactor(CategoriesModal): Remove wipNewMappingBooks feature flag

### PR DESCRIPTION
**Pull request by inesmouandjolobe**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-31067

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [X] Made sure my feature is working on web.
- [X] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)

### CODE REVIEW

File: src/features/search/helpers/categoriesHelpers/mapping-tree.test.ts [hunk 1]
```diff
@@ -1,40 +1,18 @@
...
-    expect(
-      createMappingTree(
-        mockedSubcateroriesV2Response,
-        algoliaFacets.facets as FacetData,
-        !mockedNewMappingEnabled
-      )
-    ).toEqual(expectedResult)
-  })
-
   it('should return a mapping tree for book category ROMANS_ET_LITTERATURE with FF newMappingEnabled to true', () => {
```
Comment (cedricls1) [inline src/features/search/helpers/categoriesHelpers/mapping-tree.test.ts:12]: Tu peux retirer `with FF newMappingEnabled to true`

File: src/features/search/helpers/categoriesHelpers/mapping-tree.ts [hunk 2]
```diff
@@ -114,34 +114,8 @@ function mapBookCategories(data: SubcategoriesResponseModelv2) {
...
 }
 
-export function createMappingTree(
-  data?: SubcategoriesResponseModelv2,
-  facetsData?: FacetData,
-  newMappingEnabled?: boolean
-) {
+export function createMappingTree(data?: SubcategoriesResponseModelv2, facetsData?: FacetData) {
   if (!data) return {} as MappingTree
-  /**
```
Comment (cedricls1) [inline src/features/search/helpers/categoriesHelpers/mapping-tree.ts:123]: C'était pas pratique ce comm ? ^^

**Review by cedricls1 — APPROVED** [review]

File: src/features/search/helpers/categoriesHelpers/mapping-tree.ts (hunk 2 above)
Comment (inesmouandjolobe) [reply src/features/search/helpers/categoriesHelpers/mapping-tree.ts:123]: On a déjà fait le nouveau mapping, j'avoue que ça ne me semblait plus utile mais je peux le remettre :)

File: src/features/search/helpers/categoriesHelpers/mapping-tree.test.ts (hunk 1 above)
Comment (inesmouandjolobe) [reply src/features/search/helpers/categoriesHelpers/mapping-tree.test.ts:12]: yes

**Pull request merged**